bin/benchmark_nc.py

```diff
@@ -0,0 +1,601 @@
+                                 help='Disallow to look for SSH keys in ~/.ssh/.')
+    benchmark_group.add_argument('--dir', type=str, required=False, default=None,
+                                 help='Local directory where to copy stuff. This will be mirrored to the remote system if needed (make sure to use path that will work on both).')
+    benchmark_group.add_argument('--models', type=str, nargs='+', required=False,
+                                 help='List of files (protocolbuffer) to work on. Might be a zip file.')
+    benchmark_group.add_argument('--so-model', type=str, required=False,
+                                 help='Perform one step using AOT-based .so model')
+    benchmark_group.add_argument('--wav', type=str, required=False,
+                                 help='WAV file to pass to native_client. Supply again in plotting mode to draw realine line.')
+    benchmark_group.add_argument('--alphabet', type=str, required=False,
+                                 help='Text file to pass to native_client for the alphabet.')
+    benchmark_group.add_argument('--iters', type=int, required=False, default=5,
+                                 help='How many iterations to perfom on each model.')
+    benchmark_group.add_argument('--keep', required=False, action='store_true',
+                                 help='Keeping run files (binaries & models).')
+    benchmark_group.add_argument('--csv', type=argparse.FileType('w'), required=False,
+                                 help='Target CSV file where to dump data.')
+    benchmark_group.add_argument('--binaries', type=str, required=False, default=None,
+                                 help='Specify non TaskCluster native_client.tar.xz to use')
+    plotting_group = parser.add_argument_group(title='plot', description='Plotting options')
+    plotting_group.add_argument('--dataset', action='append', nargs=2, metavar=('name','source'),
+                                help='Include dataset NAME from file SOURCE. Repeat the option to add more datasets.')
+    plotting_group.add_argument('--title', type=str, default=None,
+                                help='Title of the plot.')
+    plotting_group.add_argument('--plot', type=argparse.FileType('w'), required=False,
+                                help='Target file where to plot data. Format will be deduced from extension.')
+    plotting_group.add_argument('--size', type=str, default='800x600',
+                                help='Size (px) of the resulting plot.')
+    plotting_group.add_argument('--dpi', type=int, default=96,
+                                help='Set plot DPI.')
+    plotting_group.add_argument('--range', type=str, default=None,
+                                help='Range of model size to use. Comma-separated string of boundaries: min,max')
+    return parser.parse_args()
+
+def do_main():
+    cli_args = handle_args()
+
+    if run_as_benchmark(cli_args):
+        print('Running as benchmark')
```

Comment:
Please split the benchmarking and the plotting into separate scripts.

Reply:
I'm not too sold on that regarding the benefits :)